### ISSUE TITLE
Improve accessibility of carousel caption

### ISSUE THREAD

**Issue by midzer** [twbs/bootstrap#29494]
Hi all,
after a review of one of my sites for accessibility by a friend of mine, there was an issue for carousel caption.

According to the docs https://getbootstrap.com/docs/4.3/components/carousel/#with-captions 
```
<div class="carousel-caption d-none d-md-block">
  <h5>First slide label</h5>
  <p>Nulla vitae elit libero, a pharetra augue mollis interdum.</p>
</div>
```
we recommend putting a `<h5>` for slide label.

Most people using a screen reader navigate through the headings on the page first of all https://webaim.org/projects/screenreadersurvey8/#finding

So, when carousel comes first after an `<h1>` for example, they discover `<h5>` first, instead of jumping to the next (upcoming) `<h2>`.

I think we should replace this non-section `<h5>` by something more accessible. Styling for `font-size` could be done via `:first-child` for example.

Have a nice weekend
midzer

**Comment by XhmikosR** [comment]
We can just use our `h*` classes and this has already come up many times.

I had a patch at some point but dropped it. https://github.com/twbs/bootstrap/pull/28487

/CC @patrickhlauke

**Comment by XhmikosR** [comment]
@ffoodd can we merge this with #31908?

**Comment by ffoodd** [comment]
Hmm I'd say it's better to keep this one opened, I can open a PR next week for this. My other issue requires to check the whole design pattern.

**Comment by XhmikosR** [comment]
Just keep in mind that switching headings with our classes, will make the markup more complex, which is one of the reasons I dropped my patch. So, I'm not sure if we should change this after all :)

**Comment by ffoodd** [comment]
Might also add a callout about headings hierarchy, but as far as I know going from `h1` to `h5` is not really an issue regarding accessibility.

**Comment by patrickhlauke** [comment]
yeah skipping heading levels isn't a big issue (and normatively not necessarily an actual WCAG failure either) https://developer.paciellogroup.com/blog/2020/03/heading-off-confusion-when-do-headings-fail-wcag/

authors should use whatever heading level they feel is appropriate, really. the carousel pattern itself shouldn't mandate a particular level (which we don't).

purely for the hierarchy in the demo page, i suppose changing the `<h5>` to `<h4>` would make sense, but even then ... not a big issue at all. I'd be happy leaving it as is.

**Comment by XhmikosR** [comment]
All right, let's close it then and we revisit later, if needed.
